ksqldb-engine/src/main/java/io/confluent/ksql/engine/rewrite/DataSourceExtractor.java

```diff
@@ -131,6 +132,39 @@ public Void visitAliasedRelation(final AliasedRelation relation, final Void ctx)
       return null;
     }
 
+    private String checkAlternatives(final SourceName sourceName) {
+      final StringBuilder hint = new StringBuilder();
+
+      for (SourceName name:metaStore.getAllDataSources().keySet()) {
+        if (name.text().equalsIgnoreCase(sourceName.text())) {
+          /* if source name in metastore contains small letters, the query source name must be
+          wrapped in double quotes with the correct order of small and capital letters */
+          if (!name.text().toUpperCase().equals(name.text())
```

Comment:
I think we can simplify it a little bit and make the hint a bit more concise. 

There are 2 possibilities, and in both of them, we do not care about `sourceName` after a case-agnostic match:
1. If the matched source contains only capital letters, we should advise using it without double quotes.
2. If the matched source contains at least one lowercase letter, we should advise the correct spelling in double-quotes.

There is an additional corner case with multiple matches. In this case, we might want to simply list all options in a single sentence rather than have multiple hints. We can maybe specify whether it is a table or a stream for each option.

You might find it easier to split the method into 2 phases. In the first phase, you'd collect all matches. Then, in the second phase, you format it based on the number of matches. For example, with a single match, you can write something like `Did you mean "bAR"? Hint: wrap a source name in double quotes to make it case-sensitive.`. With multiple matches, you can write something like `Did you mean "bAR" (stream), "bar" (table), or BAR (stream)? Hint: wrap a name in double quotes to make it case-sensitive.`.